In [1]:
import numpy as np
from IPython.display import HTML
from bokeh.plotting import output_notebook, show
import genomes_dnj.lct_interval.series_plots as dm
import genomes_dnj.lct_interval.anal_series as an
import genomes_dnj.lct_interval.series_masks as sm
output_notebook(hide_banner=True)

<h3>28_434 Hierarchies</h3>
<div style="width:700px">
<p>
Four different hierarchies are selected by the series 28_434.  The associations
that distinguish those hierarchies include:
</p>
<ul>
<li>117_1685 and 123_1561
<li>117_1685 by itself
<li>117_1685 and 13_1696
<li>123_1561 and 13_1696
</ul>
<p>
These hierarchies have been formed through some combination of remodeling of series
SNP expression, recombination events, and selection through the emergence of new series.
</p>
</div>

<h3>117_1685 and 123_1561</h3>
<div style="width:700px">
<p>
The first plot shows data for chromosome samples that express series 28_434 in
association with both 117_1685 and 123_1561.  The 29 samples that
express this association include 25 that resulted from the emergence
of the series 25_27.  SNP counts show that 25 of the samples express
all of the 117_1685 SNPs.  But, 24 of the samples only express 115 of
the 123_1561 SNPs.
</p>
</div>

In [2]:
plt_obj = dm.superset_yes_no([dm.di_28_434, dm.di_117_1685, dm.di_123_1561], min_match=0.5)
plt = plt_obj.do_plot()
show(plt)

In [3]:
HTML(plt_obj.get_html())

In [4]:
an.sa_117_1685.unique_snps_per_allele(sm.am_28_434_117_1685_123_1561)

array([(114,  1), (115,  2), (116,  1), (117, 25)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

In [5]:
an.sa_123_1561.unique_snps_per_allele(sm.am_28_434_117_1685_123_1561)

array([(115, 24), (116,  2), (121,  1), (122,  1), (123,  1)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>Only 117_1685</h3>
<div style="width:700px">
<p>
This plot shows data for the subset of 28_434 chromosome samples that express
117_1685 but neither 123_1561 nor 13_1696.  The 73 chromosome samples that
express the association include 67 that have been selected through
expression of the series 6_68.  SNP counts show that 57 of the samples
express 116 of the 117_1685 SNPs.  All of the samples express large
fragments of the 123_1561 SNPs.
</div>

In [6]:
plt_obj = dm.superset_yes_no([dm.di_28_434, dm.di_117_1685], 
                             [dm.di_13_1696, dm.di_123_1561], min_match=0.5)
plt = plt_obj.do_plot()
show(plt)

In [7]:
HTML(plt_obj.get_html())

In [8]:
an.sa_117_1685.unique_snps_per_allele(sm.am_28_434_117_1685_not_123_1561_13_1696)

array([(114,  4), (115,  7), (116, 57), (117,  5)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

In [9]:
an.sa_123_1561.unique_snps_per_allele(sm.am_28_434_117_1685_not_123_1561_13_1696)

array([( 66,  1), (106,  2), (107,  4), (108,  5), (109, 44), (110, 17)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>117_1685 and 13_1696</h3>
<div style="width:700px">
<p>
This plot shows data for the 74 chromosome samples that express 28_434 along with
117_1685 and 13_1696.  These samples include 73 that express the series
22_73.  SNP counts show that 69 of the samples only express 106 of the
117_1685 SNPs.  All of the samples express large fragments of the
123_1561 SNPs.  The history that lead to this association surely included
a recombination event between a chromosome that expressed 
117_1685, 28_434, 70_166, and 49_136 with another that expressed 13_1696.
The resulting association of series became heavily overexpressed through
some kind of selective advantage introduced with the series 22_73.
</div>

In [10]:
plt_obj = dm.superset_yes_no([dm.di_28_434, dm.di_117_1685, dm.di_13_1696], 
                             [dm.di_123_1561], min_match=0.5)
plt = plt_obj.do_plot()
show(plt)

In [11]:
HTML(plt_obj.get_html())

In [12]:
an.sa_117_1685.unique_snps_per_allele(sm.am_28_434_117_1685_13_1696)

array([(105,  3), (106, 69), (107,  2)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

In [13]:
an.sa_123_1561.unique_snps_per_allele(sm.am_28_434_117_1685_13_1696)

array([(107,  8), (108, 50), (109, 15), (110,  1)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>Only 123_1561</h3>
<div style="width:700px">
<p>
This plot shows data for the 256 28_434 chromosome samples that express 123_1561 but
not 117_1685.  The samples include 251 that express the 180 SNP series
180_251.  SNP counts show that 160 of the samples express all 123_1561 SNPs.
Another 78 express 122 of them.  All of the chromosomes express significant fragments
of 117_1685.  The history that lead to this association included a recombination
event between a chromosome that expressed 123_1561, 28_434, 9_887, and
5_588 with another that expressed 13_1696.  The resulting association
of series became heavily over expressed as a result of some kind of
selective advantage associated with the series 180_251.  But, the
process that generated the 180 SNP series 180_251 series remains unclear.
</div>

In [14]:
plt_obj = dm.superset_yes_no([dm.di_28_434, dm.di_123_1561], [dm.di_117_1685], min_match=0.5)
plt = plt_obj.do_plot()
show(plt)

In [15]:
HTML(plt_obj.get_html())

In [16]:
an.sa_117_1685.unique_snps_per_allele(sm.am_28_434_123_1561_not_117_1685)

array([(57,   1), (59,   1), (60,  27), (61, 165), (62,  53), (63,   2),
       (68,   1), (92,   4), (93,   1), (98,   1)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

In [17]:
an.sa_123_1561.unique_snps_per_allele(sm.am_28_434_123_1561_not_117_1685)

array([(120,   8), (121,  10), (122,  78), (123, 160)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>123_1561 and 13_1696</h3>
<div style="width:700px">
<p>
This plot confirms that all of the 251 chromosome samples that express 180_251
also express 13_1696.  All 251 samples express significant fragements of 117_1685.
Most samples express 60, 61, or 62 SNP 117_1685 fragments.  Most samples express
the complete series of 123_1561 SNPs
</div>

In [18]:
plt_obj = dm.superset_yes_no([dm.di_28_434, dm.di_123_1561, dm.di_13_1696], [dm.di_117_1685], min_match=0.5)
plt = plt_obj.do_plot()
show(plt)

In [19]:
HTML(plt_obj.get_html())

In [20]:
an.sa_117_1685.unique_snps_per_allele(sm.am_28_434_123_1561_13_1696_not_117_1685)

array([(57,   1), (59,   1), (60,  27), (61, 165), (62,  50), (63,   2),
       (92,   4), (93,   1)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

In [21]:
an.sa_123_1561.unique_snps_per_allele(sm.am_28_434_123_1561_13_1696_not_117_1685)

array([(120,   8), (121,   9), (122,  74), (123, 160)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])